unison-share-api/src/Unison/Server/Backend.hs

```diff
@@ -811,78 +810,65 @@ mungeSyntaxText ::
   Functor g => g (UST.Element Reference) -> g Syntax.Element
 mungeSyntaxText = fmap Syntax.convertElement
 
-prettyDefinitionsBySuffixes ::
+-- | Renders a definition for the given name or hash alongside its documentation.
+prettyDefinitionsForHQName ::
+  -- | The path representing the user's current perspective.
+  -- Searches will be limited to definitions within this path, and names will be relative to
+  -- this path.
   Path ->
+  -- | The root branch to use
   Maybe (Branch.CausalHash) ->
   Maybe Width ->
+  -- | Whether to suffixify bindings in the rendered syntax
   Suffixify ->
+  -- | Runtime used to evaluate docs. This should be sandboxed if run on the server.
   Rt.Runtime Symbol ->
   Codebase IO Symbol Ann ->
-  [HQ.HashQualified Name] ->
```

Comment:
Trying to compute results for multiple terms at once makes the code more confusing and error prone, and we want to bias the printer towards each result being printed anyways, so it's likely best to do them one at a time.

Currently nobody ever calls the api with multiple terms at once anyways, and if you did it'd be tough to sort out which results are for which queries anyhow.